lib/datasource.js

```diff
@@ -348,6 +348,14 @@ DataSource.prototype.setup = function(name, settings) {
     if (settings.initialize) {
       connector = settings;
     } else if (settings.connector) {
+      // some connectors are aliases to loopback-component-storage
+      if (['object-storage'].indexOf(settings.connector) > -1) {
+        settings.provider = 'openstack';
+        settings.connector = 'loopback-component-storage';
+        settings.useServiceCatalog = true;
+        settings.useInternal = false;
+        settings.keystoneAuthVersion = 'v3';
```

Comment:
Should we allow `datasources.json` to override these settings? I.e. treat the values above only as "reasonable defaults" and allow the user/application to specify a different value of `useInternal` (or any other field) if they wish so.